addons/website_slides/controllers/main.py

```diff
@@ -699,6 +699,54 @@ def slide_send_share_email(self, slide_id, email, fullscreen=False):
     # QUIZZ SECTION
     # --------------------------------------------------
 
+    @http.route('/slides/slide/quiz/question_add_or_update', type='json', methods=['POST'], auth='user', website=True)
+    def slide_quiz_question_add_or_update(self, slide_id, question, sequence, answer_ids, id=None):
+        """Adds a new Question to an existing slide and set the completed field to False
+            in the slide.slide.partner model to make sure that the creator can take the
+            Quiz if there wasn't any to begin with. When updating, the Question is deleted
+            first before creating a new one.
+
+            :param integer slide_id: Slide ID
+            :param string question: Question Title
+            :param integer sequence: Question Sequence
+            :param array answer_ids: Array containing all the answers :
+                    [
+                        'sequence': Answer Sequence (Integer),
+                        'text_value': Answer Title (String),
+                        'is_correct': Answer Is Correct (Boolean)
+                    ]
+            :param integer id: Question ID (if it's an update)
+
+            :return: same object as received in parameter but containing the ID of the
+                     question
+        """
+        slide = request.env['slide.slide'].browse(slide_id)
+        slide_question = request.env['slide.question'].browse(id)
+        update = False
+        if slide_question.exists():
+            update = True
+            slide_question.unlink()
+        request.env['slide.slide.partner'].search([
+            ('slide_id', '=', slide_id),
+            ('partner_id', '=', request.env.user.partner_id.id)
+        ]).write({'completed': False})
+        slide_question = request.env['slide.question'].create({
+            'sequence': sequence,
+            'question': question,
+            'slide_id': slide_id,
+            'answer_ids': [(0, 0, {
+                'sequence': answer['sequence'],
+                'text_value': answer['text_value'],
+                'is_correct': answer['is_correct']
+            }) for answer in answer_ids]
+        })
+        return {'view': request.env.ref('website_slides.lesson_content_quiz_question').render({
```

Comment:
You should only return the rendered template here.

The JS should be able to know if it's an update or not using another way.